Build race condition in src\tools\illink\src\ILLink.Tasks\ILLink.Tasks.csproj

`error MSB3026: Could not copy "D:\a\_work\1\s\src\tools\illink\src\ILLink.Tasks\build\Microsoft.NET.ILLink.targets" to "D:\a\_work\1\s\artifacts\bin\ILLink.Tasks\Release\net8.0\../build\Microsoft.NET.ILLink.targets". Beginning retry 1 in 1000ms. The process cannot access the file 'D:\a\_work\1\s\artifacts\bin\ILLink.Tasks\Release\build\Microsoft.NET.ILLink.targets' because it is being used by another process.  [D:\a\_work\1\s\src\tools\illink\src\ILLink.Tasks\ILLink.Tasks.csproj::TargetFramework=net8.0]`

----
## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=361790
Build error leg or test failing: Build / windows-x86 release Runtime_Debug / Build product
Pull request: https://github.com/dotnet/runtime/pull/89911
<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "Microsoft.NET.ILLink.targets' because it is being used by another process.",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

Looking. By the way, are the runtime tests build warnings konwn?

> D:\a\_work\1\s\.dotnet\sdk\8.0.100-preview.6.23330.14\Microsoft.Common.CurrentVersion.targets(2365,5): warning MSB3246: Resolved file has a bad image, no metadata, or is otherwise inaccessible. PE image does not have metadata. PE image does not have metadata. [D:\a\_work\1\s\src\tests\nativeaot\CustomMain\CustomMain.csproj] [D:\a\_work\1\s\src\tests\build.proj]

cc @jkoritzinsky

--- a/eng/testing/linker/SupportFiles/Directory.Build.props
+++ b/eng/testing/linker/SupportFiles/Directory.Build.props
@@ -3,7 +3,7 @@
   <!-- Use live illink bits. It is necessary to both import the package props and override
        the tasks assembly, because the live package props in the build output do not use
        the same layout as the NuGet package. -->
-  <Import Project="$(ToolsILLinkDir)build/Microsoft.NET.ILLink.Tasks.props" />
+  <Import Project="$(ToolsILLinkDir)$(NetCoreAppToolCurrent)/build/Microsoft.NET.ILLink.Tasks.props" />
   <PropertyGroup>
     <ILLinkTasksAssembly>$(ToolsILLinkDir)$(NetCoreAppToolCurrent)/ILLink.Tasks.dll</ILLinkTasksAssembly>
   </PropertyGroup>

--- a/src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj
+++ b/src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj
@@ -30,9 +30,9 @@
     <!-- Note: 'build/Microsoft.NET.ILLink.targets' should not match the package name, because we don't want the targets
          to be imported by nuget. The SDK will import them in the right order. -->
     <Content Include="**\*.props;**\*.targets" Pack="true" PackagePath="%(RecursiveDir)%(Filename)%(Extension)"
-             CopyToOutputDirectory="PreserveNewest" TargetPath="../%(RecursiveDir)%(Filename)%(Extension)" />
+             CopyToOutputDirectory="PreserveNewest" TargetPath="%(RecursiveDir)%(Filename)%(Extension)" />
     <Content Include="..\ILLink.RoslynAnalyzer\build\**\*.props" Pack="true" PackagePath="build/%(RecursiveDir)%(Filename)%(Extension)"
-             CopyToOutputDirectory="PreserveNewest" TargetPath="../build/%(RecursiveDir)%(Filename)%(Extension)" />
+             CopyToOutputDirectory="PreserveNewest" TargetPath="build/%(RecursiveDir)%(Filename)%(Extension)" />
   </ItemGroup>
 
   <ItemGroup>

